tslib/src/functions-throttled/sync-results-to-sql/sync-results-data.service.spec.ts

```diff
@@ -0,0 +1,102 @@
+import * as R from 'ramda'
+import { mockCompletionCheckMessage } from './mocks/completed-check.message'
+import { SyncResultsDataService } from './sync-results.data.service'
+import { ISqlService } from '../../sql/sql.service'
+import { DBEventType } from './models'
+
+const mockQuestionData = new Map()
+mockQuestionData.set('1x1', { id: 1, factor1: 1, factor2: 2, isWarmup: false, code: 'Q001' })
+const validatedCheck = mockCompletionCheckMessage.validatedCheck
+let mockSqlService: ISqlService
+
+describe('SyncResultsDataService', () => {
+  let sut: SyncResultsDataService
+  let date: Date
+
+  beforeEach(() => {
+    mockSqlService = {
+      query: jest.fn(),
+      modify: jest.fn(),
+      modifyWithTransaction: jest.fn()
+    }
+    sut = new SyncResultsDataService(mockSqlService)
+    date = new Date()
+  })
+
+  it('calls `createNewEvent` if a new event is seen in the audit log when preparing events', async () => {
+    // setup
+    const eventTypes = new Map<string, DBEventType>()
+    eventTypes.set('TestEvent1', { id: 1, eventType: 'TestEvent1', eventDescription: '' })
+    eventTypes.set('TestEvent2', { id: 1, eventType: 'TestEvent2', eventDescription: '' })
+
+    jest.spyOn(sut as any, 'createNewEventType').mockImplementation(jest.fn())
+    jest.spyOn(sut as any, 'refreshEventTypes')
+      .mockImplementationOnce(() => {
+        // call 1: 2 event types
+        sut['eventType'] = eventTypes
+      })
+      .mockImplementationOnce(() => {
+        // call 2 - now the new event should be added
+        eventTypes.set('TestEvent3', { id: 3, eventType: 'TestEvent3', eventDescription: '' })
+        sut['eventType'] = eventTypes
```

Comment:
can we not use dot notation here?

Reply:
No.  It's a private variable.  Using object notation is an ugly work-around.